tools/archives/pyvo_integration/astronomical_archives.xml

```diff
@@ -0,0 +1,1023 @@
+
+          #end if
+
+        #end if
+
+        python '$__tool_directory__/astronomical_archives.py' '$output' '$output_csv' '$output_html' '$output_basic_html' '$output_error'
+      ]]>
+    </command>
+    <inputs>
+        <conditional name="archiveselection">
+            <param name="archive_type" type="select" label="Archive Selection">
+              <option value="registry">Query IVOA registry</option>
+              <option value="archive">Query specific IVOA archive</option>
+            </param>
+            <when value="registry">
+              <param name="keyword" type="text" label="Keyword" />
+              <param name="service_type" type="select" label="Service type">
+                <option value="TAP" selected="true">TAP: Tables</option>
+                <option value="SCS">SCS: Cone search</option>
+                <option value="SSA">SSA: Spectra</option>
+                <option value="SIA">SIA: Images</option>
+              </param>
+              <param name="wavebands" type="select" label="Wavebands">
+                <option value="all" selected="true">All</option>
+                <option value="radio">Radio</option>
+                <option value="millimeter">Millimeter</option>
+                <option value="infrared">Infrared</option>
+                <option value="optical">Optical</option>
+                <option value="uv">UV</option>
+                <option value="euv">EUV</option>
+                <option value="x-ray">X-ray</option>
+                <option value="gamma-ray">Gamma-ray</option>
+              </param>
+            </when>
+            <when value="archive">
+              <param name="archive" type="select" label="Archive">
+                <options from_data_table="astronomical_archives" />
+              </param>
+            </when>
+        </conditional>
+        <section name="query_section" title="Query selection" expanded="false">
+          <conditional name="query_selection">
+            <param name="query_type" type="select" label="ADQL Query Selection">
+              <option value="none">No specific query (first n files from archive)</option>
+              <option value="obscore_query">IVOA obscore table query builder</option>
+              <option value="raw_query">Raw ADQL query builder</option>
+            </param>
+            <when value="none"></when>
+            <when value="obscore_query">
+              <!-- Fields defined https://www.ivoa.net/documents/ObsCore/20160224/WD-ObsCore-v1.1-20160224.pdf -->
+              <param name="target_name" type="text" label="Observation target name" help="Typically name of the astronomical object observed" />
```

Comment:
can we help here with a select list?

Reply:
we could add a select list with the most common objects and then offer a free text option?

Reply:
Can we do both in one field, select+freetext? Our too superficial search did not show such an input.

Reply:
Not on one, you would need a conditional like you used below.

Reply:
In principle, would it be useful to have select-or-text input?
In select, one can search with text. Then, if not found, user has to copy what is inputed, click "input manually" and then input in another field?

Reply:
Yes, I think its a useful idea. We should do it at some point.

Reply:
And we have it already :)


```
<param name="foo" type="text" value="foo 1">
    <option value="foo 1">Foo 1 Display</option>
    <option value="foo 2">Foo 2 Display</option>
</param>
```

Reply:
Gosh Galaxy keeps surprising me.

Reply:
Imagine how we feel :) Somehow almost every time I think of some feature of idea which could work here, I find that someone already tried it in Galaxy.

Reply:
haha, that sounds like a good toot! Made my day :)